clang_tidy.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+#  Copyright (c) 2024, Intel Corporation
+#  SPDX-License-Identifier: BSD-3-Clause
+
+set -e
+
+BUILD_FODLER=${1:-"build"}
+
+if [ ! -d "$BUILD_FODLER" ]; then
+  echo "Build folder '$BUILD_FODLER' does not exist."
+  exit 1
+fi
+
+if [ ! -f "$BUILD_FODLER/compile_commands.json" ]; then
+  echo "compile_commands.json not found in '$BUILD_FODLER'."
+  exit 1
+fi
+
+FILES=$(ls                                  \
+    src/*.{cpp,h}                           \
+    src/opt/*.{cpp,h}                       \
+    *.cpp                                   \
+    builtins/*{cpp,hpp,c}                   \
+    common/*.h                              \
+)
+
+# Run clang-tidy with the compilation database from the given build folder in parallel
+NUM_CORES=$(nproc)
```

Comment:
You can use `run-clang-tidy` for parallel run:
https://github.com/llvm/llvm-project/blob/main/clang-tools-extra/clang-tidy/tool/run-clang-tidy.py